zebra-test/src/command.rs

```diff
@@ -444,6 +444,28 @@ impl<T> TestOutput<T> {
             .with_section(|| format!("{:?}", regex).header("Match Regex:"))
     }
 
+    #[instrument(skip(self))]
+    pub fn stdout_match_line(&self, regex: &str) -> Result<Vec<String>> {
+        let re = regex::Regex::new(regex)?;
+        let stdout = String::from_utf8_lossy(&self.output.stdout);
+
+        let mut matched_lines = Vec::new();
+        for line in stdout.lines() {
+            if re.is_match(line) {
+                matched_lines.push(String::from(line));
+            }
+        }
+        if !matched_lines.is_empty() {
+            return Ok(matched_lines);
+        }
+
+        Err(eyre!(
+            "stdout of command did not contain any line that matches the given regex"
+        ))
+        .context_from(self)
+        .with_section(|| format!("{:?}", regex).header("Match Regex:"))
+    }
```

Comment:
This is good code, can you split it into a separate PR and rebase it on top of #2252?

This is a low priority - if you can't make it work quickly, it's ok to leave it for later.